feat: admin note includes creator name when available

ovasdi opened the pull request:
https://artsyproduct.atlassian.net/browse/PLATFORM-4713

This builds on the [recent change](https://github.com/artsy/gravity/pull/15838) to the admin note API which exposed the `created_by` field (_user ID_) on the note.

As far as this PR, it updates the `UserAdminNoteType` to include a new field `creator` which, using the `created_by` field coming from gravity, _resolves_ using `userByIDLoader` to return the `user` who created the note.

`null` is returned in case of [legacy note](https://github.com/artsy/metaphysics/pull/4515#discussion_r1019651031) where the`created_by` field was never set during note creation:

Example shape of admin notes with `body` and `creator` selecting only the _name_ field would look like so:

![Screen Shot 2022-11-11 at 9 30 02 AM](https://user-images.githubusercontent.com/29984068/201361630-8345775e-1c77-469a-ba26-b0f303185b0c.png)

---

I _could really use some advice_ as far as how to test this change. I have been trying to modify [this realted test](https://github.com/artsy/metaphysics/blob/main/src/schema/v2/__tests__/user.test.ts#L898-L945) to include the new shape but haven't been able to get it to work.

ovasdi commented on src/schema/v2/user.ts:
```diff
@@ -43,6 +43,24 @@ export const UserAdminNoteType = new GraphQLObjectType<any, ResolverContext>({
+      resolve: async ({ created_by }, {}, { userByIDLoader }) => {
+        if (!userByIDLoader) {
+          throw new Error(
+            "Loader not found. You must supply an X-Access-Token header."
```
I am unsure as far as this message is concerned. On the `user` we have two forms of this message:

> You need to pass a X-Access-Token header to perform this action

and 

> Loader not found. You must supply an X-Access-Token header.

Which one should be used then?

ovasdi replied:
Should we use one form in this file? I am happy to update all other lines so that it's a consistent message.

mc-jones replied:
I think it is a good idea to standardize this. Would even suggest just saying `An X-Access-Token header to perform this action.`

ovasdi replied:
Made the change in this commit https://github.com/artsy/metaphysics/pull/4515/commits/0266ea806dc70c752fc62a59ee26e9b7d7e6cd3b to use the same message across the file:
> An X-Access-Token header is required to perform this action.

mc-jones replied:
😬 Put up [a33edaa](https://github.com/artsy/metaphysics/pull/4515/commits/a33edaa0a2bf53c110fdfc032cf19c9fcaf2e57b) to fix my own grammar mistake.

ovasdi commented on src/schema/v2/user.ts:
```diff
@@ -43,6 +43,24 @@ export const UserAdminNoteType = new GraphQLObjectType<any, ResolverContext>({
+        }
+
+        if (!created_by) {
+          return null // This is a legacy note, associated user is not available
```
As mentioned in the PR description, all [ResourceNote](https://github.com/artsy/gravity/blob/main/app/models/domain/resource_note.rb)(s) up to now have not set `created_by` field and so MP will return `adminName: null` for all of the 'legacy' notes.

mc-jones replied:
Good call 👍 !

joeyAghion reviewed (commented):
I _don't_ know the best answer to your questions, but regarding the schema: I think it would be more graph-y to simply have a `creator` property that loads an associated `User` type, with all the fields therein. The client can query for name or email or anything else it desires.

ovasdi commented:
> I _don't_ know the best answer to your questions, but regarding the schema: I think it would be more graph-y to simply have a `creator` property that loads an associated `User` type, with all the fields therein. The client can query for name or email or anything else it desires.

Thanks for the feedback. I initially did have it setup this way but then switched to return only the name. changed in https://github.com/artsy/metaphysics/pull/4515/commits/940d846b9bdfa8ae9aa6b9e4b29965ce87fc4168 and updated the description

ovasdi commented on src/schema/v2/user.ts:
```diff
@@ -43,6 +43,23 @@ export const UserAdminNoteType = new GraphQLObjectType<any, ResolverContext>({
     createdAt: date(({ created_at }) => created_at),
+    creator: {
+      description: "The user who created the note",
+      type: UserType,
```
I see that in [Bidder type](https://github.com/artsy/metaphysics/blob/main/src/schema/v2/bidder.ts#L28) we use `UserField.type` when resolving the `user` tho I am not clear on what the difference is between using that and `UserType` that I am using here.

ovasdi commented on src/schema/v2/user.ts:
```diff
@@ -43,6 +43,23 @@ export const UserAdminNoteType = new GraphQLObjectType<any, ResolverContext>({
+    creator: {
+      description: "The user who created the note",
+      type: UserType,
+      resolve: ({ created_by }, {}, { userByIDLoader }) => {
```
Would also like to clarify if there is a difference between passing `{}` vs `_args` vs `_` when no args are getting passed into the `resolve`. I see all three forms used across the codebase. Do we prefer one over the other(s)?

mzikherman replied:
I would probably use `_args`, but I don't think there's a difference, they're all unused after-all.

ovasdi replied:
Thanks changed to your prefered one: https://github.com/artsy/metaphysics/pull/4515/commits/cf8ee33454914507cbc7d0da96eb9386f592d8d2

mzikherman reviewed (dismissed):
Nice! Yup, rather than exposing the name or other selected attributes, exposing the `User` schema here is 👍
